python/ray/dag/dag_node_operation.py

```diff
@@ -297,17 +354,96 @@ def _build_dag_node_operation_graph(
             _add_edge(
                 graph[task_idx][_DAGNodeOperationType.WRITE],
                 graph[downstream_task_idx][_DAGNodeOperationType.READ],
+                "nccl"
+                if graph[task_idx][_DAGNodeOperationType.WRITE].requires_nccl
+                else "shm",
             )
     return graph
 
 
+def _node_repr(node: _DAGOperationGraphNode, idx: int, optimized_index: int):
+    """
+    Representation of a node in the visualization of the execution schedule.
+
+    Args:
+        node: The node to be represented.
+        idx: The index of the node in the execution schedule.
+        optimized_index: The index of the node in the optimized execution schedule.
+    """
+    return str(node) + f" {idx},{optimized_index}"
+
+
+def _visualize_execution_schedule(
+    actor_to_execution_schedule: Dict[
```

Comment:
Why do we need the original schedule?

Reply:
To easily compare the old and new schedule and identify the differences, this is quite useful in debugging.